sql/core/src/main/scala/org/apache/spark/sql/parquet/ParquetTypes.scala

```diff
@@ -229,11 +231,15 @@ private[parquet] object ParquetTypesConverter extends Logging {
     case LongType => Some(ParquetTypeInfo(ParquetPrimitiveTypeName.INT64))
     case TimestampType => Some(ParquetTypeInfo(ParquetPrimitiveTypeName.INT96))
     case DecimalType.Fixed(precision, scale) if precision <= 18 =>
-      // TODO: for now, our writer only supports decimals that fit in a Long
       Some(ParquetTypeInfo(ParquetPrimitiveTypeName.FIXED_LEN_BYTE_ARRAY,
         Some(ParquetOriginalType.DECIMAL),
         Some(new DecimalMetadata(precision, scale)),
         Some(BYTES_FOR_PRECISION(precision))))
+    case DecimalType.Fixed(precision, scale) =>
+      Some(ParquetTypeInfo(ParquetPrimitiveTypeName.BINARY,
```

Comment:
Using `BINARY` here conforms to Parquet format spec. But according to the spec, `FIXED_LENGTH_BYTE_ARRAY` with different length can also be used to store decimals with different precisions. From the perspective of storage efficiency, `FIXED_LENGTH_BYTE_ARRAY` is probably more preferable, since `BINARY` has variable length and needs 4 extra bytes to encode the length (before being encoded and compressed).

Another benefit here is that we can just unify cases for precision <= 18 and precision > 18.